data/error_policies/virtual_function_non_define.cas

```diff
@@ -0,0 +1,5 @@
+virtual resource foo_parent {
+	virtual fn foo_func(domain source) {}
+}
+
+virtual resource foo inherits foo_parent {} // Illegal non-inheritance
```

Comment:
I guess people may want to automatically inherit a virtual `foo_func()`  in `foo` but I think indeed that making them explicit is cleaner, similar to Rust traits.

Reply:
Actually, I think the test as written is wrong here.  What I wrote in the doc is:

> Unlike concrete types, virtual types inherit virtual functions from their parents.

So in this case, the virtual should be automatically inherited.  However, for a *concrete* type, the function would need to be explicitly declared or derived.  I think this is all spelled out clearly in doc/TE.md "Type Inheritance" section, but to try to recap concisely and cleanly here:

* A concrete type should inherit its parents' concrete functions automatically, and a virtual type should inherit its parents' virtual functions automatically.  The exception to both of these is a name conflict.
* For virtual functions not automatically inherited, the child must provide an implementation either explicitly or using the @derive macro.
* For concrete functions, the child may provide an implementation either explicitly or using the derive macro, regardless of whether they were automatically inherited.

In light of that, I believe this test is incorrect in declaring "foo" as virtual, and will break once the above is fully implemented, so I'll update it to remove the virtual keyword.

Reply:
I haven't implemented the function derivation portion yet, so most of this isn't working.  I'd like to add the full set of test cases along with the @derive annotation which I'll be working on in June.